### PR TITLE
fix(app): Address protocols landing design feedback

### LINKED ISSUE #10208: 6.0 Feedback: Design QA: T1 Protocols Landing Page #10153
### Overview

Round 1 Feedback for Protocol Details page

### Current Behavior

n/a

### Expected Behavior

Link to view the annotated screenshot and side by side comparison to the intended design [on Figma](https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=11331%3A330817)

- [X] (1) Update styling of button to reflect our secondary style (no fill, update all states)
- [x] (2) Missing 'sort by' functionality
- [x] (3) Need 4px padding in between each card
- [X] (4) should be our p, semibold in dark-black #16212D
- [x] (5) should b our "link/secondary/grey/external" link styling which relies on our label font styling. Make sure to have all the states (currently missing them)
- [x] (6) center align the external svg icon and the text. Icon should be 12px, which is the same line height as the label text. 
- [x] (7) Need to align last updated text with design. Make sure it's follows the "styling/last-updated" style we have defined and in the correct location (fixed to the right hand side of the card)
- [x] (8) All text should be our dark-black #16212D, except for some the h6 headers, last updated date,  and links. 
- [x] (9) These h6's should be our dark-grey/enabled #707075, letter-spacing: 1%
- [x] (10) Can't figure out the flex box settings, but when the viewport is at 800px width, these columns should be able to hold the max number of characters without going onto two rows. (probably the "P1000 Single-Channel GEN2"). Not sure if it's because 'last updated' isn't tucked to the right? 
- [x] (11) The opened menu should be 4px below the menu overflow icon
- [x] (12) Menu overflow icon behavior [Loom video explaining](https://www.loom.com/share/3c49f4c275874210ae6dd2048a712d54)(please contact me if you have questions)
- [ ] (13) SVG adjustment. [Loom video explaining](https://www.loom.com/share/0c675ba03b80459f8c69db76e965e77b) (please contact me when you hit this, i think we'll need to work close together to solve)
- [x] (14) Update size of the spinner
- [x] (15) Update colour of the 'loading data...'
- [x] (16) Update to our h1 style, keeping it mind the colour should also be our 'dark-black'
- [x] (17) Update to our p style, make sure it's 'dark-black' colour
- [x] (18) Update to our primary button style 
- [x] (19) Update 'file_drop_zone' to our updated component in the design system 
- [x] (20) This entire section should be centered within the inner frame (see the original design to the left in Figma)
- [x] (21) should be the same amount of margin we have after the protocol cards when there's content (40px)

Screen Shot 2022-05-04 at 12 02 59 PM
Screen Shot 2022-05-04 at 2 41 49 PM

### Steps To Reproduce

_No response_

### Operating system

_No response_

### Robot setup or anything else?

_No response_

### LINKED ISSUE #10555: 6.0 Feedback:  T1 Protocol Card SVG Fix
### Overview

This is to fix a spacing issue around the SVG on the protocol cards. 

### Current Behavior

[Link to the figma frame ](https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=11331%3A330817)with annotations

[Loom video link](https://www.figma.com/file/2xjSLHXquNsKvC1mV9NMjr/Milestone-2?node-id=955%3A144959) explaining the issue 

- [ ] (13) Think I'll need to work 1-on-1 with a dev to understand what's going on here. Loom video should explain most of it! Just DM me when you start working on this. 

Screen Shot 2022-05-31 at 2 25 22 PM

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Operating system

_No response_

### Robot setup or anything else?

_No response_

### PR DESCRIPTION
**Pull request by koji**
# Overview
This will address feedback in #10767 and fix #10208 which is the original design feedback for protocols landing page.
In addition, this will fix #10555

Screen Shot 2022-07-06 at 1 44 38 PM

[width 600px]
Screen Shot 2022-07-06 at 1 44 27 PM

Screen Shot 2022-07-06 at 12 15 45 PM

* in terms of (15), need to get details from Mel. This will be added when she is back.
Opened a new issue  for No.15, #11000

# Changelog
- address design feedback in #10767

# Review requests
- [ ] Would like to get advice to remove the purple space in the video. 

https://user-images.githubusercontent.com/474225/177001941-7447348f-5e5b-41d6-81cd-985bff96c045.mov

- [ ] design feedback is addressed

# Risk assessment
low

### CODE REVIEW

File: app/src/organisms/ProtocolAnalysisFailure/index.tsx [hunk 1]
```diff
@@ -50,7 +50,12 @@ export function ProtocolAnalysisFailure(
     dispatch(analyzeProtocol(protocolKey))
   }
   return (
-    <Banner type="error" marginBottom={SPACING.spacing4}>
+    <Banner
```
Comment (jerader) [inline app/src/organisms/ProtocolAnalysisFailure/index.tsx:53]: Looks like the very edge of the temperature module is cut off:

Screen Shot 2022-07-05 at 2 48 07 PM

File: app/src/organisms/ProtocolsLanding/ProtocolList.tsx [hunk 2]
```diff
@@ -28,11 +31,22 @@ import { MenuItem } from '../../atoms/MenuList/MenuItem'
 import type { StoredProtocolData } from '../../redux/protocol-storage'
 import type { ProtocolSort } from './hooks'
 
+const SORT_BY_STYLE = css`
```
Comment (jerader) [inline app/src/organisms/ProtocolsLanding/ProtocolList.tsx]: these states make the protocol cards look really nice

**Review by jerader — COMMENTED** [review]
Just left a small comment on the deck thumbnail svg viewbox.

Additionally, this might be out of scope of this PR but the Protocol Card is not very responsive. If you shrink the viewport, the `Updated` column gets cut off
Screen Shot 2022-07-05 at 2 52 56 PM

File: app/src/organisms/ProtocolsLanding/ProtocolCard.tsx [hunk 3]
```diff
@@ -75,6 +77,7 @@ export function ProtocolCard(props: ProtocolCardProps): JSX.Element | null {
       position="relative"
```
Comment (brenthagen) [inline app/src/organisms/ProtocolsLanding/ProtocolCard.tsx:82]: line 73: `marginBottom={SPACING.spacing2}`

File: app/src/organisms/ProtocolsLanding/ProtocolList.tsx [hunk 4]
```diff
@@ -91,18 +105,23 @@ export function ProtocolList(props: ProtocolListProps): JSX.Element | null {
...
+            >
               {t('labware_landing:sort_by')}
             </StyledText>
             <Flex
               flexDirection={DIRECTION_ROW}
               alignItems={ALIGN_CENTER}
-              backgroundColor={COLORS.medGrey}
               borderRadius={BORDERS.radiusSoftCorners}
               marginLeft={SPACING.spacing3}
+              css={SORT_BY_STYLE}
```
Comment (brenthagen) [inline app/src/organisms/ProtocolsLanding/ProtocolList.tsx]: this should also have `cursor="pointer"`

File: app/src/organisms/ProtocolsLanding/ProtocolCard.tsx (hunk 3 above)
Comment (koji) [reply app/src/organisms/ProtocolsLanding/ProtocolCard.tsx:82]: Good catch!

File: app/src/organisms/ProtocolsLanding/ProtocolList.tsx (hunk 4 above)
Comment (koji) [reply app/src/organisms/ProtocolsLanding/ProtocolList.tsx]: Thank you, Brent. This made me notice the current `sort by` button issue.

File: app/src/molecules/DeckThumbnail/index.tsx [hunk 5]
```diff
@@ -40,10 +40,13 @@ export function DeckThumbnail(props: DeckThumbnailProps): JSX.Element {
...
 
   return (
+    // PR #10488 changed size
+    // revert the height
+    // Note add offset 18px to right and left
     <RobotWorkSpace
       deckLayerBlocklist={deckSetupLayerBlocklist}
       deckDef={deckDef}
-      viewBox="-90 -40 560 500"
+      viewBox="-75 -20 586 480"
```
Comment (koji) [inline app/src/molecules/DeckThumbnail/index.tsx:49]: Got Jethary's feedback and changed x from `-68` to `-75`.

https://github.com/Opentrons/opentrons/pull/10993#discussion_r914110955


[current viewBox settings]
Screen Shot 2022-07-06 at 10 32 43 AM

File: app/src/organisms/ProtocolsLanding/useGetElementProperty.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+
+type DOMRectProperty = keyof Omit<DOMRect, 'toJSON'>;
+interface props {
+    getElementProperty: (property: DOMRectProperty) => number;
+}
+
+export const useGetElementProperty = <T extends HTMLElement>(
```
Comment (brenthagen) [inline app/src/organisms/ProtocolsLanding/useGetElementProperty.ts]: consider naming this `useGetElementDOMRectProperty` for clarity

**Review by brenthagen — APPROVED** [review]
✨

**Review by jerader — APPROVED** [review]
🦕

**Pull request merged**